x-pack/test/security_api_integration/tests/session_concurrent_limit/cleanup.ts

```diff
@@ -181,7 +180,10 @@ export default function ({ getService }: FtrProviderContext) {
       await setTimeoutAsync(500);
       const basicSessionCookieThree = await loginWithBasic(testUser);
 
-      expect(await getNumberOfSessionDocuments()).to.be(3);
+      log.debug('Waiting for all sessions to be persisted...');
+      await retry.tryForTime(20000, async () => {
+        expect(await getNumberOfSessionDocuments()).to.be(3);
+      });
```

Comment:
__note:__ after reviewing the errors from the failed runs, it appears it was just a timing issue, hence the `retry`.